### PR TITLE
Added optional batch normalization mode parameter.

### PR DESCRIPTION
**Pull request by jmhessel**
### What and why?
I added an optional parameter for the batch normalization mode in ResNet50 (see #2). If one wanted to use ResNet50 as a shared layer, the Keras BatchNormalization mode must be set to 2. I believe that this change allows for that. The default is mode = 0, so this should not break any existing code.

### Potential Downsides
- It's a very specific use case, and might not be worth merging; this could be something folks could add on their own
- I am not 100% sure that the ImageNet weight initialization still makes sense for mode = 2, though I suspect it does.

### CODE REVIEW

**Comment by fchollet** [comment]
> I am not 100% sure that the ImageNet weight initialization still makes sense for mode = 2, though I suspect it does.

I think that's a somewhat significant issue. Inference performance would likely be much lower.

Overall I'd rather let users fork the model and make whatever change they need. Same if they need to add regularization.
